vta/hardware/chisel/src/main/scala/core/Store.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package vta.core
+
+import chisel3._
+import chisel3.util._
+import vta.util.config._
+import vta.shell._
+
+class Store(debug: Boolean = false)(implicit p: Parameters) extends Module {
```

Comment:
Is there a better pattern for threading debug state instead of putting a per-module parameter?

Reply:
I could make them "implicit" and add them "hierarchically" here in the [core config file](https://github.com/vegaluisjose/tvm/blob/nVTA-pr/vta/hardware/chisel/src/main/scala/core/Configs.scala). What do you think?

Reply:
That might be nice, so we can turn on/off debugging in one place, is there a way to key config by module? so we don't have to write boiler plate for each one? I could look into it after the tutorial if we want to put cycles else where.

Reply:
Yeah, we can discuss this later to see what is the best way to go.